mava/wrappers/gym.py

```diff
@@ -0,0 +1,299 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import traceback
+import warnings
+from multiprocessing import Queue
+from multiprocessing.connection import Connection
+from typing import Any, Callable, Dict, Optional, Tuple, Union
+
+import gymnasium
+import numpy as np
+from gymnasium import spaces
+from gymnasium.vector.utils import write_to_shared_memory
+from numpy.typing import NDArray
+
+# Filter out the warnings
+warnings.filterwarnings("ignore", module="gymnasium.utils.passive_env_checker")
+
+
+class GymWrapper(gymnasium.Wrapper):
+    """Wrapper for gym environments."""
```

Comment:
This line is likely too long, but if you can add in this comment so we know it works for rware

```suggestion
    """Base wrapper for multi-agent gym environments.
    
    This wrapper works out of the box for RobotWarehouse. See `GymLBFWrapper` for how it can be modified to work for other environments.
    """
```